### DISCUSSION TITLE
Functions that take variadic tuple doesn't recognize empty tuple

### DISCUSSION THREAD

**Discussion by sylee957** [microsoft/pyright#4873] [Q&A] [answered]
Given the `int, ... -> str` functions:

```python3
from typing import Callable


def a(var: tuple[()]) -> str:
    raise NotImplementedError


def b(var: tuple[int, int]) -> str:
    raise NotImplementedError


def c(var: tuple[int, int, int]) -> str:
    raise NotImplementedError
```

I recognize that the empty tuple function does not pass the type check

```
f: Callable[[tuple[int, ...]], str]
f = a # Does not pass type check
f = b
f = c
```

While the empty tuple passes the type checking.
```
x: tuple[int, ...]
x = ()
x = (1,)
x = (1, 2)
```
I'm using pyright 1.1.301

**Comment by erictraut** [comment, marked as answer]
Yes, I agree this is an inconsistency. Pyright should not emit an error in this case. I've created a [bug report](https://github.com/microsoft/pyright/issues/4876) for tracking purposes.
